### DISCUSSION TITLE
Is there a way to seed data into Unleash?

### DISCUSSION THREAD

**Discussion by jejacks0n** [Unleash/unleash#2677] [Q&A] [answered]
I'm working on an adapter that lets [ActiveExperiment](https://github.com/jejacks0n/activeexperiment-unleash) use Unleash as its rollout and metrics service.

I want to write some tests that connect to a real Unleash server, so the plan in CI is to spin up a docker container with Unleash running (which is pretty easy) -- and to use the admin API to add specific feature toggles, and variants. The code I have for that looks like this:

```ruby
Unleash::Util::Http.post(
  URI("#{Unleash.configuration.url_stripped_of_slash}/admin/projects/default/features"),
  {
    "type": "experiment",
    "name": "MyExperimentToggle",
    "description": "This is a feature toggle for ActiveExperiment",
    "impressionData": true
  }.to_json
)
```

Now, this works fine too -- but requires an admin token. So at a minimum it seems like I have to seed in an admin token somehow, so then I went looking, and I can connect to the database and export/import data directly, but I'm curious if there's any official tooling that I'm not seeing for this?

**Comment by jejacks0n** [comment, marked as answer]
I found it -- there's an `INIT_ADMIN_API_TOKENS` environment variant that can be used, which is perfect, and will let me use some of the code I already have.

**Reply by thomasheartman** [reply]
Yeah, spot on! `INIT_ADMIN_API_TOKENS` is there for cases like this exactly. There's also equivalent variables for client
(`INIT_CLIENT_API_TOKENS`) and front-end (`INIT_FRONTEND_API_TOKENS`) tokens if you need that 😄
